src/MediaForm.php

```diff
@@ -94,12 +94,12 @@ public function save(array $form, FormStateInterface $form_state) {
   /**
    * {@inheritdoc}
    */
-  public function validate(array $form, FormStateInterface $form_state) {
-    parent::validate($form, $form_state);
+  public function validateForm(array &$form, FormStateInterface $form_state) {
+    parent::validateForm(&$form, $form_state);
     /** @var \Drupal\media_entity\MediaInterface $entity */
     $entity = $this->buildEntity($form, $form_state);
     /** @var \Drupal\media_entity\MediaInterface $entity_unchanged */
-    if ($entity->id()) {    
+    if ($entity->id()) {
```

Comment:
Changed stuff removed. Created follow-up for conversion of type validations to a constraint: https://www.drupal.org/node/2529178